sklearn/datasets/_openml.py

```diff
@@ -911,7 +911,7 @@ def fetch_openml(
     if parser == "auto":
         parser = "liac-arff" if return_sparse else "pandas"
```

Comment:
shouldn't we use `liac-arff` if `pandas` is not installed instead of raising?

Reply:
You can refer to the discussion with @ogrisel https://github.com/scikit-learn/scikit-learn/pull/21938#discussion_r796845328

Basically, since the two parsers behave slightly differently, changing depending on the libraries installed will be difficult to debug and it fails. I recall such a bug in `pandas` with `numexpr` is installed or not, and this is not a good thing.

Reply:
Also `fetch_openml` with default arguments on a dense dataset will already raise an exception in scikit-learn 1.0 because `as_frame="auto"` arlready requires pandas if the dataset is dense.